extensions-core/druid-basic-security/src/main/java/org/apache/druid/security/basic/BasicSecurityDruidModule.java

```diff
@@ -168,6 +211,33 @@ public static BasicAuthorizerCacheNotifier createAuthorizerCacheNotifier(final I
     );
   }
 
+  /**
+   * Returns the instance provided either by a config property or coordinator-run class or default class.
```

Comment:
I think the semantics of this method are too complex. Maybe you can extract a method in fewer places, but they are simpler, e. g. they accept non-null parameters.

Reply:
Config based properties might be null so anyway there will be null-checks